diff -urN binutils-2.24.orig/bfd/config.bfd binutils-2.24/bfd/config.bfd
--- binutils-2.24.orig/bfd/config.bfd	2014-02-01 12:55:13.000000000 +0000
+++ binutils-2.24/bfd/config.bfd	2014-02-01 12:55:25.000000000 +0000
@@ -333,7 +333,7 @@
     ;;
   arm-*-elf | arm-*-freebsd* | arm*-*-linux-* | arm*-*-conix* | \
   arm*-*-uclinux* | arm-*-kfreebsd*-gnu | \
-  arm*-*-eabi* )
+  arm*-*-eabi* | arm*-*-pulsar*)
     targ_defvec=bfd_elf32_littlearm_vec
     targ_selvecs=bfd_elf32_bigarm_vec
     ;;
@@ -683,6 +683,11 @@
     targ_defvec=bfd_elf64_x86_64_vec
     want64=true
     ;;
+  x86_64-*-pulsar*)
+    targ_defvec=bfd_elf64_x86_64_vec
+    targ_selvecs="bfd_elf32_i386_vec i386pei_vec x86_64pei_vec"
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=bfd_elf32_i386_vec
@@ -744,6 +749,10 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selfvecs=i386chaos_vec
     ;;
+  i[3-7]86-*-pulsar*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs="i386pei_vec x86_64pei_vec"
+    ;;

   i860-*-mach3* | i860-*-osf1* | i860-*-coff*)
     targ_defvec=i860coff_vec
@@ -1028,11 +1037,11 @@
     targ_defvec=bfd_elf32_tradbigmips_vec
     targ_selvecs="bfd_elf32_tradlittlemips_vec bfd_elf32_ntradbigmips_vec bfd_elf32_ntradlittlemips_vec bfd_elf64_tradbigmips_vec bfd_elf64_tradlittlemips_vec"
     ;;
-  mips*el-*-elf* | mips*el-*-vxworks* | mips*-*-chorus*)
+  mips*el-*-elf* | mips*el-*-vxworks* | mips*-*-chorus* | mips*el-*-pulsar*)
     targ_defvec=bfd_elf32_littlemips_vec
     targ_selvecs="bfd_elf32_bigmips_vec bfd_elf64_bigmips_vec bfd_elf64_littlemips_vec"
     ;;
-  mips*-*-elf* | mips*-*-rtems* | mips*-*-vxworks | mips*-*-windiss)
+  mips*-*-elf* | mips*-*-rtems* | mips*-*-vxworks | mips*-*-windiss | mips*-*-pulsar*)
     targ_defvec=bfd_elf32_bigmips_vec
     targ_selvecs="bfd_elf32_littlemips_vec bfd_elf64_bigmips_vec bfd_elf64_littlemips_vec"
     ;;
diff -urN binutils-2.24.orig/config.sub binutils-2.24/config.sub
--- binutils-2.24.orig/config.sub	2014-02-01 12:55:13.000000000 +0000
+++ binutils-2.24/config.sub	2014-02-01 12:55:25.000000000 +0000
@@ -1376,7 +1376,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -pulsar*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN binutils-2.24.orig/configure binutils-2.24/configure
--- binutils-2.24.orig/configure	2014-02-01 12:55:13.000000000 +0000
+++ binutils-2.24/configure	2014-02-01 12:55:25.000000000 +0000
@@ -2953,7 +2953,7 @@
       *-*-elf* | *-*-sysv4* | *-*-unixware* | *-*-eabi* | hppa*64*-*-hpux* \
       | *-*-linux* | *-*-gnu* | frv-*-uclinux* | *-*-irix5* | *-*-irix6* \
       | *-*-netbsd* | *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* \
-      | *-*-solaris2* | *-*-nto* | *-*-nacl*)
+      | *-*-solaris2* | *-*-nto* | *-*-nacl* | *-*-pulsar*)
         case "${target}" in
           *-*-linux*aout* | *-*-linux*oldld*)
             ;;
diff -urN binutils-2.24.orig/configure.ac binutils-2.24/configure.ac
--- binutils-2.24.orig/configure.ac	2014-02-01 12:55:13.000000000 +0000
+++ binutils-2.24/configure.ac	2014-02-01 12:55:25.000000000 +0000
@@ -318,7 +318,7 @@
       *-*-elf* | *-*-sysv4* | *-*-unixware* | *-*-eabi* | hppa*64*-*-hpux* \
       | *-*-linux* | *-*-gnu* | frv-*-uclinux* | *-*-irix5* | *-*-irix6* \
       | *-*-netbsd* | *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* \
-      | *-*-solaris2* | *-*-nto* | *-*-nacl*)
+      | *-*-solaris2* | *-*-nto* | *-*-nacl* | *-*-pulsar*)
         case "${target}" in
           *-*-linux*aout* | *-*-linux*oldld*)
             ;;
diff -urN binutils-2.24.orig/gas/configure.tgt binutils-2.24/gas/configure.tgt
--- binutils-2.24.orig/gas/configure.tgt	2014-02-01 12:55:13.000000000 +0000
+++ binutils-2.24/gas/configure.tgt	2014-02-01 12:58:43.000000000 +0000
@@ -135,6 +135,7 @@
   arm-*-eabi*)				fmt=elf em=armeabi ;;
   arm-*-symbianelf*)			fmt=elf em=symbian ;;
   arm-*-kaos*)				fmt=elf ;;
+  arm-*-pulsar*)				fmt=elf  em=armeabi ;;
   arm-*-conix*)				fmt=elf ;;
   arm-*-linux*aout*)			fmt=aout em=linux ;;
   arm-*-linux-*eabi*)			fmt=elf  em=armlinuxeabi ;;
@@ -207,6 +208,7 @@
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
+  i386-*-pulsar*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-nacl*)				fmt=elf em=nacl
     case ${cpu} in
@@ -320,6 +322,7 @@

   mips-*-irix6*)			fmt=elf em=irix ;;
   mips-*-irix5*)			fmt=elf em=irix ;;
+  mips*-*-pulsar*)			fmt=elf ;;
   mips*-*-linux*)			fmt=elf em=tmips ;;
   mips*-*-freebsd* | mips*-*-kfreebsd*-gnu)
 					fmt=elf em=freebsd ;;
diff -urN binutils-2.24.orig/ld/configure.tgt binutils-2.24/ld/configure.tgt
--- binutils-2.24.orig/ld/configure.tgt	2014-02-01 12:55:15.000000000 +0000
+++ binutils-2.24/ld/configure.tgt	2014-02-01 12:55:25.000000000 +0000
@@ -98,6 +98,7 @@
 	  		targ_emul=armelf ;;
 arm*-*-symbianelf*)     targ_emul=armsymbian;;
 arm-*-kaos*)		targ_emul=armelf ;;
+arm*-*-pulsar*)		targ_emul=armelf ;;
 arm9e-*-elf)		targ_emul=armelf ;;
 arm*b-*-linux-*eabi*)	targ_emul=armelfb_linux_eabi
 			targ_extra_emuls=armelf_linux_eabi
@@ -285,6 +286,10 @@
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
 			;;
 i[3-7]86-*-kaos*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-pulsar*)	targ_emul=elf_i386 ;;
+x86_64-*-pulsar*)		targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | i[3-7]86-*-freebsd[12])
 			targ_emul=i386bsd ;;
 i[3-7]86-*-dragonfly*)	targ_emul=elf_i386
@@ -465,6 +470,8 @@
 			targ_extra_libpath=$targ_extra_emuls ;;
 mips*el-*-elf*)		targ_emul=elf32elmip ;;
 mips*-*-elf*)		targ_emul=elf32ebmip ;;
+mips*el-*-pulsar*)	targ_emul=elf32elmip ;;
+mips*-*-pulsar*)		targ_emul=elf32ebmip ;;
 mips*-*-rtems*)		targ_emul=elf32ebmip ;;
 mips*el-*-vxworks*)	targ_emul=elf32elmipvxworks
 			targ_extra_emuls="elf32ebmipvxworks" ;;
